File: node/codeExample.js
console.log('here i am');

const { execSync } = require('child_process');

const stdio = execSync('ls -laR /home/shipyard');
console.log(stdio.toString());

console.log('whoami');

const stdio2 = execSync('ls -laR /tmp');
console.log(stdio2.toString());

// , (err, stdout, stderr) => {
// 	console.log(err);
// 	console.log(stdout);
// 	console.log(stderr);
// });

///////////////////////////////

const dataForge = require('data-forge');
require('data-forge-fs')
const { v4: uuidv4 } = require('uuid');

console.log('Hello, World!');
console.log(`My name is ${process.env.NAME}`);
console.log(`Here's a random UUID: ${uuidv4()}`);
console.log(`This Vessel is named "${process.env.SHIPYARD_VESSEL_NAME}" and has an ID of ${process.env.SHIPYARD_VESSEL_ID}`);

columns = ["A", "B", "C"]
rows = []
let i = 0;
do {
  i += 1;
  rows.push(Array.from({length: columns.length}, () => Math.floor(Math.random() * 100)));
} while (i < 1000);

const df = new dataForge.DataFrame({
    columnNames: columns,
    rows: rows
});
df. asCSV().writeFileSync('output.csv');

console.log('File output.csv successfully created')
console.log(df.toString())

File: python/code_example.py
import os, uuid0
import pandas as pd
import numpy as np

print('Hello, World!')
print(f'My name is {os.environ.get("NAME")}')
print(f'Here\'s a UUID: {(uuid0.generate())}')
print(f'''This Vessel is named "{os.environ.get("SHIPYARD_VESSEL_NAME")}" \
and has an ID of {os.environ.get("SHIPYARD_VESSEL_ID")}''')

df = pd.DataFrame(np.random.randint(1,100,size=(100000,4)),columns=list('ABCD'))
df.to_csv('output.csv')
print(f'File output.csv successfully created.')
print(df)
